R/batch-chat.R

```diff
@@ -0,0 +1,333 @@
+#' Submit multiple chats in one batch
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' `batch_chat()` and `batch_chat_structured()` currently only work with
+#' [chat_openai()] and [chat_anthropic()]. They use the OpenAI and Anthropic
+#' batch APIs which allow you to submit multiple requests simultaneously.
+#' The results can take up to 24 hours to complete, but in return you pay 50%
+#' less than usual (but note that ellmer doesn't include this discount in
+#' its pricing metadata). If you want to get results back more quickly, or
+#' you're working with a different provider, you may want to use
+#' [parallel_chat()] instead.
+#'
+#' Since batched requests can take a long time to complete, `batch_chat()`
+#' requires a file path that is used to store information about the batch so
+#' you never lose any work. You can either set `wait = FALSE` or simply
+#' interrupt the waiting process, then later, either call `batch_chat()` to
+#' resume where you left off or call `batch_chat_completed()` to see if the
+#' results are ready to retrieve. Once you have the results, you can delete
+#' the file.
```

Comment:
Maybe mention that you could also keep the file to cache the results?

And maybe also that `batch_chat()` stores the results in the file (rather than retrieving them anew from the provider) so as long as you have the file you can access the results. Re-running a script and using cached `batch_chat()` results each time as Tomasz mentioned sounds both like a feature and a footgun.